File: commons/rs/models/src/article_count.rs
use serde_derive::{Serialize, Deserialize};

#[derive(Serialize, Deserialize)]
pub struct ArticleCount {
    pub total: i64,
    pub cnt_contained_search_term: i64,
    pub cnt_not_contained_search_term: i64,
}

impl ArticleCount {
    pub fn new(
        total: i64,
        cnt_contained_search_term: i64,
        cnt_not_contained_search_term: i64,
    ) -> Self {
        Self {
            total,
            cnt_contained_search_term,
            cnt_not_contained_search_term,
        }
    }
}

File: backend/py_rest/py_rest/models/article_count.py
from dataclasses import dataclass


@dataclass
class ArticleCount:
    total: int
    cnt_contained_search_term: int
    cnt_not_contained_search_term: int

    @classmethod
    def from_response(cls, response):
        print(response.total)
        print(response.cnt_contained_search_term)
        print(response.cnt_not_contained_search_term)
        return cls(
            total=response.total,
            cnt_contained_search_term=response.cnt_contained_search_term,
            cnt_not_contained_search_term=response.cnt_not_contained_search_term,
        )
